user-data-store-rest-client/src/main/java/com/wultra/security/userdatastore/UserDataStoreRestClientConfiguration.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * User Data Store
+ * Copyright (C) 2024 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.security.userdatastore;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+
+import java.time.Duration;
+
+/**
+ * Configuration of PowerAuth REST client.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ *
+ */
+public class UserDataStoreRestClientConfiguration {
```

Comment:
It may use Lombok `Getter` and `Setter`. But the main question is, why do we introduce a specific class and not reuse the existing `RestClientConfiguration`?

Reply:
Could you please add `responseTimeout`, `maxIdleTime`, and `maxLifeTime` properties, if you keep your own configuration class?

Reply:
Good point, I will switch to `RestClientConfiguration`.